# Семинар 1: Введение в Python и условные конструкции

## Краткое повторение первой недели

## Типы данных: Строки (текст) - STR (STRING)

Вас взяли на стажировку в учебный офис ФКМД, и дали задание создать адреса корпоративных почт для первокурсников. Обычно такая почта состоит из первой буквы имени, первой буквы отчества и фамилии, а потом добавляется домен @edu.hse.ru . Так как умный человек должен быть по-хорошему ленив, то вы решили задействовать для этого Python и сделать программу, которая будет запрашивать Имя, Отчество и Фамилию латиницей, а потом выдавать адрес. Как можно написать такую программу?

In [ ]:
# здесь мы можем решить эту супер-задачу

In [ ]:
print('Моё имя:')

Моё имя:


In [ ]:
type('Маргарита')

str

In [ ]:
print('Первая'+'пара')

Перваяпара


In [ ]:
print('У меня '+ 'проблема'*20)

У меня проблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблемапроблема


In [ ]:
print(1 + 2)  
print('1' + '2')  
print(str(1) + str(2))  

3
12
12


In [ ]:
'Программирование'[0]

'П'

In [ ]:
'Программирование'[2]

'о'

In [ ]:
'Программирование'[-1]

'е'

In [ ]:
'Программирование'[-3]

'н'

In [ ]:
'Программирование'[-20]

IndexError: ignored

# Числа

Вы решили сделать в своей квартире небольшой ремонт и переклеить обои в каждой из 3 комнат (спальня, гостиная и коридор). Рулон флизелиновых обоев под покраску имеет длину 25 метров и ширину 1.06 метра. Считать вручную на листочке не очень интересно, поэтому необходимо написать программу, которая считает ширину и длину помещения и выдаст вам количество рулонов обоев (помните, что рулоны продаются только целиком)

In [ ]:
# здесь мы можем решить эту супер-задачу

### Типы данных: Целые числа - INT (INTEGER)

In [ ]:
print(type(5))

<class 'int'>


In [ ]:
print(4 + 2)
print(4 - 2)

6
2


In [ ]:
print(4 * 2)
print(4 / 2)
print(3 % 2)

8
2.0
1


In [ ]:
print(3 ** 2)
print(3 ** (1 / 2))

9
1.7320508075688772


## Типы данных: Вещественные числа - FLOAT

In [ ]:
type(1.5)

float

In [ ]:
print(2,6 / 2) 
print(2, 6 / 2)  
print(2.6 / 2)  

2 3.0
2 3.0
1.3


In [ ]:
print(9 / 2)  
print(9 // 2)  
print(9 % 2)  

4.5
4
1


In [ ]:
print(0.1+0.2)

0.30000000000000004


In [ ]:
1.543**1000

2.3223413936165768e+188

Такая запись расшифровывается следующим образом и называется научным видом числа:

$ n \cdot e+188 = n \cdot 10^{188} $

## Типы данных: Логические переменные - boolean

Вы решили сделать чат-бот, который будет тренировать грамотность ваших однокурсников и коллег (журналистам надо быть грамотными!). Вам нужно проверять каждое введенное слово: совпадает оно с эталонным написанием или нет? Проще всего это делать с помощью логических выражений. Попробуйте для начала написать код, который будет проверять корректность написания слова "оттоманка"(это такой диванчик)

In [ ]:
# здесь мы можем решить эту супер-задачу

In [ ]:
print(type(False))

<class 'bool'>


In [ ]:
print(type('True'))

<class 'str'>


In [ ]:
print(34 == 30+4)

True


In [ ]:
print(35 == 30-5)

False


## Ошибки в Python и что с ними делать

Иногда написанный код не работает. Так бывает. Это не страшно.

Например, я хочу вывести на экран текст следующего вида:
```
День недели: суббота.
Моё имя: Маргарита
Мой возраст: 28
```

Я написала свой код в ячейке ниже.

Выполните следующую ячейку (казалось бы, что может пойти не так?):

In [ ]:
print('День недели: суббота.')
print('Моё имя: ' + 'Маргарита')
print('Мой возраст: ' + 28)

День недели: суббота.
Моё имя: Маргарита


TypeError: can only concatenate str (not "int") to str

## Переменные и ввод данных

In [ ]:
number = 99
my_name = 'Маргарита'

In [ ]:
print(number*my_name)

МаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргаритаМаргарита


## Время приключений!(заданий)

![](https://media.giphy.com/media/fm4WhPMzu9hRK/giphy.gif)

**Задание 1** Вы стали учебным ассистентом по курсу программирования на языке Python (вот это повезло!). Вам необходимо написать программу, которая будет считывать две переменных: ФИО и оценку студента и выводить текст: "Уважаемый ФИО, за курс по Python вы получили ОЦЕНКА!"

Пример ввода:
    
Журналистов Питон Медиакомович

10

Пример вывода:

Уважаемый Журналистов Питон Медиакомович, за курс по Python вы получили 10!

In [ ]:
# ваш супер код

**Задание 2** Вы решили следить за своим режимом сна и создать программу, которая будет вычислять нужное время пробуждения в зависимости от времени засыпания и необходимой длительности сна. Напишите программу, которая будет считывать три целых числа : час засыпания, минуты засыпания, длительность сна (в минутах) и выводить время пробуждения в формате **ЧЧ:ММ**. Будем считать, что вы всегда ложитесь спать после полуночи, так как делаете домашки по питону.

Пример ввода:
    
1 <br>
20 <br>
350 <br>

Пример вывода:

7:10

In [ ]:
#ваш еще более крутой супер кот (или код)

## Пора узнать нечто новое: условные конструкции:

Иногда бывает нужно выполнять какую-то часть программы только при выполнении определенных условий.

Формальная схема:

![](http://krivaksin.ru/wp-content/uploads/2015/11/2015-11-11-11-51-19-Microsoft-PowerPoint-Prezentatsiya1.png)

Синтаксис в питоне: <br />
**if** условие1:<br />
&emsp; команды<br />
**elif** условие2:<br />
&emsp; команды<br />
**elif** условие3:<br />
&emsp; команды<br />
**else**: <br />
&emsp; команды<br />

Все команды пишутся с отступом.

Неформальный, но такой жизненный пример:

<a href="https://ibb.co/drfFvxs"><img src="https://i.ibb.co/bP6TVDj/Untitled-Frame-1-13.jpg" alt="Untitled-Frame-1-13" border="0"></a>

**Пример с оценкой:**

if оценка>=8:

        получаем автомат
elif оценка>=4:

        отдыхаем
else:

        готовимся к экзамену

### Операторы сравнения

Как правило, в качестве проверяемого условия используется результат вычисления одного из следующих операторов сравнения:

<center> 
    <table>
        <tr>
            <th>Логическое выражение </th>
            <th> Значение </th>
        </tr> 
        <tr><td>&lt;</td><td> Первый операнд меньше второго </td></tr>
        <tr><td>&gt;</td><td> Первый операнд больше второго</td></tr>
        <tr><td>&lt;=</td><td> Меньше или равно</td></tr>
        <tr><td>&gt;=</td><td> Больше или равно</td></tr>
        <tr><td>==</td><td> Два операнда равны </td></tr>
        <tr><td>!=</td><td> Два операнда не равны</td></tr>
    </table>
</center>

Чтобы записать сложное логическое выражение, часто бывают нужны связки "и", "или" и "не".

В Питоне они обозначаются как **and**, **or** и **not** соответственно.

Операции **and** и **or** являеются бинарными, т.е. должны быть записаны между операндами, например x < 3 or y > 2

Операция **not** - унарная и должна быть записана перед единственным своим операндом. Например not 3

## Пример

In [ ]:
mark = int(input())
if mark >= 8:
    print("Получаем автомат")
elif mark >=4:
    print("На пересдачу не идем, спокойно отдыхаем")
else:
    print('Грозит пересдача, усиленно готовимся')

In [ ]:
mark = int(input())
if mark >= 8:
    print("Получаем автомат")
else:
    if mark >=4:
        print("На пересдачу не идем, спокойно отдыхаем")
    else:
        print('Грозит пересдача, усиленно готовимся')

### Задача 1

Вы пытаетесь понять, положена ли вам ПГАС за успешную учебу. В этом году ПГАС за учебу дают всем, чей GPA превышает 9.5 .Напишите программу, которая считывает GPA и выводит в зависимости от его значения "Ура, Вы получили пегас!" или "Не повезло". Если введенное число будет больше 10, выведите "Вы совершенно не умеете считать".

In [ ]:
#пишите код

### Задача 2

В соответствии с анализом популярности известного всем паблика "хайер скул оф мемс", ваш семинарист выяснил, что успешным является мем, который набрал как минимум 60 лайков и 10 репостов. Пользователь вводит отдельно количество лайков и количество репостов. Выведите фразу "Ура, топовый мем", если мем соответствует параметру успешного и "Чувство юмора следует подкачать" в обратном случае

In [ ]:
#пишите код

### Задача 3

Для данного числа n<100 закончите фразу “У меня на этой неделе...” одним из возможных продолжений: “n дедлайн”, "n дедлайна”, “n дедлайнов”, правильно склоняя слово “дедлайн”.

In [ ]:
#пишите код

### Задача 4

Попробуйте решить задачу про будильник, но уже для случая, когда вы можете лечь до полуночи (т.е., может начаться новый день)

In [ ]:
#пишите код